sstmac/hardware/nic/nic.h

```diff
@@ -119,7 +119,7 @@ class NIC : public ConnectableSubcomponent
     LogP
   } Port;
 
-  virtual std::string toString() const override = 0;
+  std::string toString() const override = 0;
```

Comment:
What warning did this generate?

Reply:
Without looking at the commit that triggered this it was almost certainly the clang-tidy fixit behavior. To be fair adding virtual to a method that is override is redundant.  Also actually if this method is already pure virtual in the base class what is the point of even having it in NIC?

Reply:
There is no parent virtual method in integrated core. Probably what we should do is completely delete the line here. Then let's make an `SSTIntegratedSubcomponent` that provides the top-level virtual function. Same problem applies to memory_model and a few other things.  We want all subcomponents to be forced to give themselves string names.

Reply:
This should be a separate issue.

Reply:
That's fine if warnings are clean.